File: Easy/905. Sort Array By Parity/905. Sort Array By Parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function (nums) {
    let left = 0; // Pointer for even numbers

    for (let right = 0; right < nums.length; right++) {
        // If the current number is even, swap it with the left pointer
        if (nums[right] % 2 === 0) {
            const temp = nums[left];
            nums[left] = nums[right];
            nums[right] = temp;
            left++; // Move the left pointer to the right
        }
    }

    return nums;
};

File: Easy/905. Sort Array By Parity/905. Sort Array By Parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int left = 0; // Pointer for even numbers

        for (int right = 0; right < nums.length; right++) {
            // If the current number is even, swap it with the left pointer
            if (nums[right] % 2 == 0) {
                int temp = nums[left];
                nums[left] = nums[right];
                nums[right] = temp;
                left++; // Move the left pointer to the right
            }
        }

        return nums;
    }
}
